kotest-property/src/commonMain/kotlin/io/kotest/property/Gen.kt

```diff
@@ -66,8 +79,18 @@ abstract class Arb<out A> : Gen<A>() {
    /**
     * Edgecase values for this type A.
     */
+   @Deprecated("implement one value at a time using generateEdgecase(rs) or use Arb<T>.withEdgecases(...) instead")
    abstract fun edgecases(): List<A>
 
+   /**
+    * Returns a random edgecase value using the supplied random source
+    */
+   open fun generateEdgecase(rs: RandomSource): A =
```

Comment:
I'm not so sure this should be a non empty list.
With this implementation, if you ask for an edge case and the arb doesn't provide any, we return a random value, which is perhaps misleading given the name of the function?.
Perhaps this can just return null and then the generate functionality can then defer to the normal samples method.

Reply:
Sounds good, I can make that change if you think it's better. However, as a library user I don't know if I'd necessarily agree with the design. Firstly, unlike option null doesn't nest so you can't really encode e.g. edgecases of `Arb<A?>`. `Arb<Option<A>>` is still fine but you'd get `Option<A>?` in the edgecases..? As well the type signature of `generateEdgecases(rs): A?` doesn't really imply empty edgecases. It suggests that the edgecase may be null in one iteration and not null in another. I'd personally not put such design if i were given a choice. Rather I'd pursue something like `Edgecases<A>` which is a sealed class and it may either be empty, or probabilistic. I.e `edges(): Edgecases<A>`.

```kotlin
sealed class Edgecases<A> {
    object Empty : Edgecases<Nothing>()
    data class Probabilistic(generate: (RandomSource) -> A) : Edgecases<A>()
}
```

This way user would know straight up that the edgecases are empty with no ambiguity. 

In terms of building edgecases we might do something like

```kotlin
edgecasesOf("foo", "bar")

// Or
edgecases { rs -> ... }
```

Out of curiosity i can't remember why we decided not to pursue the approach of having a seperate sealed class?